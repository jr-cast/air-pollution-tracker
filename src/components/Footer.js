const Footer = () => (
  <div className="footerCont">
    Copyright &copy; | Jose R. Casta√±os
  </div>
);

export default Footer;
