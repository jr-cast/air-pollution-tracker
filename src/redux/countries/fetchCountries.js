const FETCH_API = 'air-pollution-tracker/countries/FETCH_API';
const initialState = [];
const API_KEY = 'b748e1c2a10434b969fb13060689a0a8';
const baseUrl = (lat, lon, API_KEY) => `http://api.openweathermap.org/data/2.5/air_pollution?lat=${lat}&lon=${lon}&appid=${API_KEY}`;

export const fetchCountries = () => async (dispatch) => {
  const connect = await fetch(baseUrl(24.8091, 107.3940, API_KEY));
  const res = await connect.json().then((dataObject) => dataObject);
  dispatch({
    type: FETCH_API,
    payload: res,
  });
};

export default function reducerCountries(state = initialState, action) {
  switch (action.type) {
    case FETCH_API:
      return action.payload;
    default:
      return state;
  }
}
