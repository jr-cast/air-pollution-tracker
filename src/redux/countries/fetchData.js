const FETCH_NA = 'air-pollution-tracker/cities/FETCH_NA';
const FETCH_CA = 'air-pollution-tracker/cities/FETCH_CA';
const FETCH_SA = 'air-pollution-tracker/cities/FETCH_SA';
const FETCH_TC = 'air-pollution-tracker/cities/FETCH_TC';
const initialState = [];

export const fetchNa = () => async (dispatch) => {
  const connect = await fetch('https://restcountries.com/v3.1/region/ame');
  const res = await connect.json().then((dataObject) => dataObject);
  const dataArray = res.map((item) => {
    const container = {
      country: item.name.common,
      population: item.population,
      capital: item.capital[0],
      region: item.subregion,
      flags: item.flags.png,
      latlng: item.latlng,
    };
    return container;
  });
  const northAmerica = dataArray.filter((item) => item.region === 'North America');
  dispatch({
    type: FETCH_NA,
    payload: northAmerica,
  });
};

export const fetchCa = () => async (dispatch) => {
  const connect = await fetch('https://restcountries.com/v3.1/region/ame');
  const res = await connect.json().then((dataObject) => dataObject);
  const dataArray = res.map((item) => {
    const container = {
      country: item.name.common,
      population: item.population,
      capital: item.capital[0],
      region: item.subregion,
      flags: item.flags.png,
      latlng: item.latlng,
    };
    return container;
  });
  const centralAmerica = dataArray.filter((item) => item.region === 'Central America');
  dispatch({
    type: FETCH_CA,
    payload: centralAmerica,
  });
};

export const fetchSa = () => async (dispatch) => {
  const connect = await fetch('https://restcountries.com/v3.1/region/ame');
  const res = await connect.json().then((dataObject) => dataObject);
  const dataArray = res.map((item) => {
    const container = {
      country: item.name.common,
      population: item.population,
      capital: item.capital[0],
      region: item.subregion,
      flags: item.flags.png,
      latlng: item.latlng,
    };
    return container;
  });
  const southAmerica = dataArray.filter((item) => item.region === 'South America');
  dispatch({
    type: FETCH_SA,
    payload: southAmerica,
  });
};

export const fetchTc = () => async (dispatch) => {
  const connect = await fetch('https://restcountries.com/v3.1/region/ame');
  const res = await connect.json().then((dataObject) => dataObject);
  const dataArray = res.map((item) => {
    const container = {
      country: item.name.common,
      population: item.population,
      capital: item.capital[0],
      region: item.subregion,
      flags: item.flags.png,
      latlng: item.latlng,
    };
    return container;
  });
  const theCaribbean = dataArray.filter((item) => item.region === 'Caribbean');
  dispatch({
    type: FETCH_TC,
    payload: theCaribbean,
  });
};

export function reducerNa(state = initialState, action) {
  switch (action.type) {
    case FETCH_NA:
      return action.payload;
    default:
      return state;
  }
}

export function reducerCa(state = initialState, action) {
  switch (action.type) {
    case FETCH_CA:
      return action.payload;
    default:
      return state;
  }
}

export function reducerSa(state = initialState, action) {
  switch (action.type) {
    case FETCH_SA:
      return action.payload;
    default:
      return state;
  }
}

export function reducerTc(state = initialState, action) {
  switch (action.type) {
    case FETCH_TC:
      return action.payload;
    default:
      return state;
  }
}
