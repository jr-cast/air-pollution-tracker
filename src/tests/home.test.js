import {
  screen, render,
} from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import { reducerNa } from '../redux/countries/fetchData';
import america from '../images/america.png';
import Footer from '../components/Footer';

const mockStore = () => {
  const preloadedState = {
    northAmerica: [
      {
        country: 'Canada',
        population: 38005238,
        capital: 'Ottawa',
        region: 'North America',
        flags: 'https://flagcdn.com/w320/ca.png',
        latlng: [
          60,
          -95
        ]
      },
      {
        country: 'United States',
        population: 329484123,
        capital: 'Washington, D.C.',
        region: 'North America',
        flags: 'https://flagcdn.com/w320/us.png',
        latlng: [
          38,
          -97
        ]
      },
    ],
  };
  return configureStore({
    preloadedState,
    reducer: {
      northAmerica: reducerNa,
    }
  })
}

const mockHome = () => {
  const fetchNa = [
    {
      country: 'Canada',
      population: 38005238,
      capital: 'Ottawa',
      region: 'North America',
      flags: 'https://flagcdn.com/w320/ca.png',
      latlng: [
        60,
        -95
      ]
    },
    {
      country: 'United States',
      population: 329484123,
      capital: 'Washington, D.C.',
      region: 'North America',
      flags: 'https://flagcdn.com/w320/us.png',
      latlng: [
        38,
        -97
      ]
    },
  ];

  const na = fetchNa;

  return (
    <div className="wrapper" >
      <div className="homeContainer">
        <img id="america" src={america} alt="american-continent" />
        <div className="mainTitle">
          <h1>Cities</h1>
          <p>Total: 56</p>
        </div>
      </div>
      <p id="stats">Select capital city by region</p>
      <div className="regionCont">
        <div role="button" tabIndex={0} className="na">{`North America ${na.length} capital cities`}</div>
      </div>
      <Footer />
    </div >
  );
}

describe('Testing the Home component', () => {
  test('Renders mocked home component properly', () => {
    const tree = renderer.create(mockHome()).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('When app runs north america region is displayed correctly', () => {
    render(<Provider store={mockStore()}>{mockHome()}</Provider>);
    expect(screen.getByAltText("american-continent")).toBeTruthy();
  });

  test('When app runs north america 2 capital cities are loaded correctly', () => {
    render(<Provider store={mockStore()}>{mockHome()}</Provider>);
    expect(screen.getAllByText("North America 2 capital cities")).toBeTruthy();
  });
});
