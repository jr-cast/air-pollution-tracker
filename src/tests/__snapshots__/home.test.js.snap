// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="wrapper"
>
  <div
    className="homeContainer"
  >
    <img
      alt="american-continent"
      id="america"
      src="america.png"
    />
    <div
      className="mainTitle"
    >
      <h1>
        Cities
      </h1>
      <p>
        Total: 56
      </p>
    </div>
  </div>
  <p
    id="stats"
  >
    Select capital city by region
  </p>
  <div
    className="regionCont"
  >
    <div
      className="na"
      role="button"
      tabIndex={0}
    >
      North America 2 capital cities
    </div>
  </div>
  <div
    className="footerCont"
  >
    Copyright © | Jose R. Castaños
  </div>
</div>
`;

exports[`Testing the Home component Renders mocked home component properly 1`] = `
<div
  className="wrapper"
>
  <div
    className="homeContainer"
  >
    <img
      alt="american-continent"
      id="america"
      src="america.png"
    />
    <div
      className="mainTitle"
    >
      <h1>
        Cities
      </h1>
      <p>
        Total: 56
      </p>
    </div>
  </div>
  <p
    id="stats"
  >
    Select capital city by region
  </p>
  <div
    className="regionCont"
  >
    <div
      className="na"
      role="button"
      tabIndex={0}
    >
      North America 2 capital cities
    </div>
  </div>
  <div
    className="footerCont"
  >
    Copyright © | Jose R. Castaños
  </div>
</div>
`;
